### PR TITLE
Codecov analysis

### PR DESCRIPTION
**Pull request by jhnlsn**
# Description of the issue
Agent is not reporting code coverage report

# Description of changes
Adding coverage report and uploading to codecov

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
_Describe what tests you have done._

### CODE REVIEW

File: .github/workflows/build.yml
Comment (SaxyPandaBear) [inline .github/workflows/build.yml:66]: I don't think that we need to/should upload test coverage on a pull request. Shouldn't that only be when we merge to master?

Comment (jhnlsn) [reply .github/workflows/build.yml:62]: It's not setup yet, but it will eventually allow for blocking merge if test coverage goes down.

File: .github/workflows/build.yml
Comment (SaxyPandaBear) [inline .github/workflows/build.yml:33]: Any particular reason we have to specify a stricter version of Go here?

File: .github/workflows/build.yml
Comment (SaxyPandaBear) [reply .github/workflows/build.yml:62]: Works for me.

File: .github/workflows/build-test-windows.yml
Comment (SaxyPandaBear) [inline .github/workflows/build-test-windows.yml:33]: I feel like I'm missing something, but why is there no build step for windows? Will there be anything to cache if we just run `go test`? Should this retain some of the flags present in the Makefile for the test command, like `-v -failfast`?

File: .github/workflows/build-test-windows.yml
Comment (SaxyPandaBear) [inline .github/workflows/build-test-windows.yml:5]: The main branch for this repo is still `master`. Smells like a copy pasta mistake

File: .github/workflows/build-test-windows.yml
Comment (SaxyPandaBear) [inline .github/workflows/build-test-windows.yml:7]: Do we do version tags such that we would trigger this Action on a tag?

File: .github/workflows/build-test-windows.yml
Comment (SaxyPandaBear) [inline .github/workflows/build-test-windows.yml:8]: Should this match the existing workflow? with
```
types: [opened, synchronize, reopened, ready_for_review]
``` 
for pull requests?

File: .github/workflows/build.yml
Comment (SaxyPandaBear) [inline .github/workflows/build.yml:72]: Is there anything we should add to the README/docs about the artifacts that we are archiving?

File: .github/workflows/build.yml
Comment (jhnlsn) [reply .github/workflows/build.yml:33]: 1.15.15 is the starting point.  `~` says < 1.16.0.  So all patch versions of 1.15.x above starting point.

File: .github/workflows/build-test-windows.yml
Comment (jhnlsn) [reply .github/workflows/build-test-windows.yml:33]: I noticed the same thing as you did.  windows took a long time and it wasn't clear why.  I looked at the open telemetry build and they broke out windows into it's ok workflow.  Doing the same here brought down build time, using cache... i have no idea why... it looks identical.

> Should this retain some of the flags present in the Makefile for the test command, like -v -failfast?
-v is verbose, don't think we need verbose unless something is broken

> I feel like I'm missing something, but why is there no build step for windows?
this runs on a windows runner, nothing extra is needed.

File: .github/workflows/build-test-windows.yml
Comment (jhnlsn) [reply .github/workflows/build-test-windows.yml:5]: I have no idea how this is even working!  wow...

File: .github/workflows/build-test-windows.yml
Comment (jhnlsn) [reply .github/workflows/build-test-windows.yml:7]: copypasta

File: .github/workflows/build.yml
Comment (SaxyPandaBear) [reply .github/workflows/build.yml:33]: The existing go version was `^1.15`, which unless I'm mistaken, could pull something >=1.16.0

Are we targeting specific releases of Go? Do we need to have compatibility for versions other than 1.15?

File: .github/workflows/build-test-windows.yml
Comment (SaxyPandaBear) [reply .github/workflows/build-test-windows.yml:5]: I think it's because the `pull_request` caught this PR. The push would be after this gets merged

File: .github/workflows/build.yml
Comment (jhnlsn) [reply .github/workflows/build.yml:72]: I think I'm going to drop the archive, we already do the nightly builds.  Not really needed right now.

**Comment by jhnlsn** [comment]
Random flaky test failing only on mac runner.

Unable to repro on local mac.

Investigating

```
=== RUN   TestOffsetDoneCallBack
2021/11/03 05:50:28 D! [inputs.tail] Seeked /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tailsrctest-173177128.log - &{Offset:0 Whence:0}
2021/11/03 05:50:29 I! [inputs.tail] Re-opening truncated file /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tailsrctest-173177128.log ...
2021/11/03 05:50:29 D! [inputs.tail] Successfully reopened truncated /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tailsrctest-173177128.log
2021/11/03 05:50:38 I! [inputs.tail] Re-opening truncated file /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tailsrctest-173177128.log ...
2021/11/03 05:50:38 D! [inputs.tail] Successfully reopened truncated /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tailsrctest-173177128.log
    tailersrc_test.go:217: Failed to parse offset: strconv.Atoi: parsing "": invalid syntax, from ''
    tailersrc_test.go:221: Wrong offset 0 is written to state file, after truncate and write shorter logs expecting 505
2021/11/03 05:50:40 W! [inputs.tail] Stopping tail as file no longer exists: /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tailsrctest-173177128.log
--- FAIL: TestOffsetDoneCallBack (12.59s)
FAIL
```

File: .github/workflows/build.yml
Comment (jhnlsn) [reply .github/workflows/build.yml:33]: yes, right now we are only targeting 1.15.x.  `^` was incorrectly being used.

File: .github/workflows/build-test-windows.yml
Comment (jhnlsn) [reply .github/workflows/build-test-windows.yml:5]: I'll change it, but also longer term need to move to main branch as default.

File: plugins/inputs/logfile/tailersrc_test.go
Comment (sethAmazon) [inline plugins/inputs/logfile/tailersrc_test.go:11]: Are the changes in this file related to code cov is so how?

File: Makefile
Comment (sethAmazon) [inline Makefile:124]: I'm not sure if unit test cache will work after this change. The build will still be cached so I do not think it will matter too much. @SaxyPandaBear are unit tests cached currently in this repo?

Comment (SaxyPandaBear) [reply Makefile:124]: I don't think it cached tests before either. Pretty sure it only cached build artifacts but I didn't pay too much attention

File: plugins/inputs/logfile/tailersrc_test.go
Comment (jhnlsn) [reply plugins/inputs/logfile/tailersrc_test.go:11]: No, this is a flakey test that uses timers to test asynchronous code... I'm almost tempted to just rip it out but for now just increased the timers instead.

File: Makefile
Comment (jhnlsn) [reply Makefile:124]: Test cache can work without -v or are you referencing the extra coverage addition?

File: plugins/inputs/logfile/tailersrc_test.go
Comment (SaxyPandaBear) [inline plugins/inputs/logfile/tailersrc_test.go:224]: Do we need to log this? If `err` is not nil, then the line below will fail out the test.

**Review by SaxyPandaBear — APPROVED** [review]
LGTM - but I think we should track flaky tests and fix them as they come to keep that issue at bay.

**Comment by jhnlsn** [comment]
@SaxyPandaBear I agree that we need to keep eyes on flakey tests.  Having codecov analysis will help us moving forward track coverage as tests are added/removed.

**Review by adam-mateen — APPROVED** [review]
I pulled changes and ran `make test` on my macbook - Tests passed.
There was _another_ flaky test - 
```
2021/11/09 11:37:26 The state file  for /var/folders/7c/rx6r0v6n44j4pwnc_77gt46w0000gs/T/465203861 does not exist: stat : no such file or directory
2021/11/09 11:37:27 The state file  for /var/folders/7c/rx6r0v6n44j4pwnc_77gt46w0000gs/T/2010428618 does not exist: stat : no such file or directory
2021/11/09 11:37:27 The state file  for /var/folders/7c/rx6r0v6n44j4pwnc_77gt46w0000gs/T/174237886 does not exist: stat : no such file or directory
2021/11/09 11:37:27 Reading from offset 9323 in /tmp/logfile.log
2021/11/09 11:37:27 W! [inputs.tail] Stopping tail as file no longer exists: /var/folders/7c/rx6r0v6n44j4pwnc_77gt46w0000gs/T/465203861
2021/11/09 11:37:27 W! [inputs.tail] Stopping tail as file no longer exists: /var/folders/7c/rx6r0v6n44j4pwnc_77gt46w0000gs/T/2010428618
2021/11/09 11:37:28 W! [inputs.tail] Stopping tail as file no longer exists: /var/folders/7c/rx6r0v6n44j4pwnc_77gt46w0000gs/T/174237886
2021/11/09 11:37:29 The state file  for /var/folders/7c/rx6r0v6n44j4pwnc_77gt46w0000gs/T/tmp2_4185061179 does not exist: stat : no such file or directory
2021/11/09 11:37:33 W! [inputs.tail] Stopping tail as file no longer exists: /var/folders/7c/rx6r0v6n44j4pwnc_77gt46w0000gs/T/tmp2_4185061179
2021/11/09 11:37:33 The state file  for /var/folders/7c/rx6r0v6n44j4pwnc_77gt46w0000gs/T/3291901238 does not exist: stat : no such file or directory
2021/11/09 11:37:33 The state file  for /var/folders/7c/rx6r0v6n44j4pwnc_77gt46w0000gs/T/1806985181 does not exist: stat : no such file or directory
2021/11/09 11:37:33 W! [inputs.tail] Stopping tail as file no longer exists: /var/folders/7c/rx6r0v6n44j4pwnc_77gt46w0000gs/T/3291901238
--- FAIL: TestLogsFileRemove (1.06s)
    logfile_test.go:354: tailerSrc should have stopped after tile is removed
FAIL
```
It passed on the 2nd run.

Here is the result showing code coverage:
```
adamym@88665a370425 amazon-cloudwatch-agent % make test                                         
CGO_ENABLED=0 go test -coverprofile coverage.txt -failfast ./awscsm/... ./cfg/... ./cmd/... ./handlers/... ./internal/... ./logger/... ./logs/... ./metric/... ./plugins/... ./profiler/... ./tool/... ./translator/...
ok  	github.com/aws/amazon-cloudwatch-agent/awscsm	0.363s	coverage: 68.1% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/awscsm/csm	0.747s	coverage: 12.4% of statements
?   	github.com/aws/amazon-cloudwatch-agent/awscsm/csm/csmiface	[no test files]
ok  	github.com/aws/amazon-cloudwatch-agent/awscsm/sdkmetricsdataplane	1.030s	coverage: 20.8% of statements
?   	github.com/aws/amazon-cloudwatch-agent/awscsm/sdkmetricsdataplane/sdkmetricsdataplaneiface	[no test files]
ok  	github.com/aws/amazon-cloudwatch-agent/cfg/agentinfo	0.263s	coverage: 90.6% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/cfg/aws	3.481s	coverage: 8.5% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/cfg/commonconfig	0.279s	coverage: 9.1% of statements
?   	github.com/aws/amazon-cloudwatch-agent/cfg/envconfig	[no test files]
ok  	github.com/aws/amazon-cloudwatch-agent/cfg/migrate	0.620s	coverage: 87.2% of statements
?   	github.com/aws/amazon-cloudwatch-agent/cmd/amazon-cloudwatch-agent	[no test files]
ok  	github.com/aws/amazon-cloudwatch-agent/cmd/amazon-cloudwatch-agent-config-wizard	0.768s	coverage: 57.4% of statements
?   	github.com/aws/amazon-cloudwatch-agent/cmd/config-downloader	[no test files]
ok  	github.com/aws/amazon-cloudwatch-agent/cmd/config-translator	1.268s	coverage: 0.0% of statements
?   	github.com/aws/amazon-cloudwatch-agent/cmd/start-amazon-cloudwatch-agent	[no test files]
?   	github.com/aws/amazon-cloudwatch-agent/handlers	[no test files]
?   	github.com/aws/amazon-cloudwatch-agent/internal	[no test files]
?   	github.com/aws/amazon-cloudwatch-agent/internal/containerinsightscommon	[no test files]
?   	github.com/aws/amazon-cloudwatch-agent/internal/csm	[no test files]
ok  	github.com/aws/amazon-cloudwatch-agent/internal/ecsservicediscovery	0.970s	coverage: 55.4% of statements
?   	github.com/aws/amazon-cloudwatch-agent/internal/httpclient	[no test files]
ok  	github.com/aws/amazon-cloudwatch-agent/internal/k8sCommon/k8sclient	1.371s	coverage: 50.8% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/internal/k8sCommon/k8sutil	0.718s	coverage: 33.3% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/internal/k8sCommon/kubeletutil	1.270s	coverage: 55.6% of statements
?   	github.com/aws/amazon-cloudwatch-agent/internal/logscommon	[no test files]
ok  	github.com/aws/amazon-cloudwatch-agent/internal/mapWithExpiry	1.589s	coverage: 100.0% of statements
?   	github.com/aws/amazon-cloudwatch-agent/internal/models	[no test files]
ok  	github.com/aws/amazon-cloudwatch-agent/internal/publisher	19.101s	coverage: 93.2% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/internal/retryer	4.545s	coverage: 100.0% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/internal/structuredlogscommon	0.781s	coverage: 95.3% of statements
?   	github.com/aws/amazon-cloudwatch-agent/internal/tls	[no test files]
ok  	github.com/aws/amazon-cloudwatch-agent/logger	1.243s	coverage: 22.2% of statements
?   	github.com/aws/amazon-cloudwatch-agent/logs	[no test files]
?   	github.com/aws/amazon-cloudwatch-agent/metric/distribution	[no test files]
ok  	github.com/aws/amazon-cloudwatch-agent/metric/distribution/regular	1.188s	coverage: 77.6% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/metric/distribution/seh1	0.906s	coverage: 73.1% of statements
?   	github.com/aws/amazon-cloudwatch-agent/plugins	[no test files]
ok  	github.com/aws/amazon-cloudwatch-agent/plugins/inputs/awscsm	0.895s	coverage: 25.1% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/plugins/inputs/cadvisor	0.987s	coverage: 2.0% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/plugins/inputs/cadvisor/extractors	0.957s	coverage: 58.8% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/plugins/inputs/demo	0.989s	coverage: 66.7% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/plugins/inputs/k8sapiserver	1.042s	coverage: 22.4% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/plugins/inputs/logfile	40.106s	coverage: 75.4% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/plugins/inputs/logfile/globpath	0.793s	coverage: 90.4% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/plugins/inputs/logfile/tail	4.243s	coverage: 39.0% of statements
?   	github.com/aws/amazon-cloudwatch-agent/plugins/inputs/logfile/tail/watch	[no test files]
ok  	github.com/aws/amazon-cloudwatch-agent/plugins/inputs/prometheus_scraper	1.029s	coverage: 37.3% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/plugins/inputs/statsd	0.643s	coverage: 75.9% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/plugins/inputs/statsd/graphite	0.934s	coverage: 72.1% of statements
?   	github.com/aws/amazon-cloudwatch-agent/plugins/inputs/win_perf_counters	[no test files]
?   	github.com/aws/amazon-cloudwatch-agent/plugins/inputs/windows_event_log	[no test files]
?   	github.com/aws/amazon-cloudwatch-agent/plugins/inputs/windows_event_log/wineventlog	[no test files]
ok  	github.com/aws/amazon-cloudwatch-agent/plugins/outputs/awscsm	0.953s	coverage: 49.9% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/plugins/outputs/awscsm/metametrics	1.389s	coverage: 82.4% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/plugins/outputs/awscsm/providers	2.203s	coverage: 71.9% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/plugins/outputs/cloudwatch	94.900s	coverage: 82.9% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/plugins/outputs/cloudwatchlogs	12.424s	coverage: 59.3% of statements
?   	github.com/aws/amazon-cloudwatch-agent/plugins/outputs/console	[no test files]
?   	github.com/aws/amazon-cloudwatch-agent/plugins/parsers	[no test files]
ok  	github.com/aws/amazon-cloudwatch-agent/plugins/parsers/awscsm	0.510s	coverage: 58.3% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/plugins/parsers/emf	0.575s	coverage: 93.3% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/plugins/processors/delta	0.490s	coverage: 90.0% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/plugins/processors/ec2tagger	7.377s	coverage: 87.2% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/plugins/processors/ecsdecorator	0.226s	coverage: 48.6% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/plugins/processors/emfProcessor	0.228s	coverage: 91.5% of statements
?   	github.com/aws/amazon-cloudwatch-agent/plugins/processors/k8sdecorator	[no test files]
ok  	github.com/aws/amazon-cloudwatch-agent/plugins/processors/k8sdecorator/stores	0.984s	coverage: 69.9% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/plugins/processors/k8sdecorator/structuredlogsadapter	0.360s	coverage: 44.4% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/profiler	0.869s	coverage: 76.5% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/tool/data	0.765s	coverage: 81.0% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/tool/data/config	0.731s	coverage: 81.4% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/tool/data/config/logs	0.972s	coverage: 82.1% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/tool/data/config/metric	0.324s	coverage: 100.0% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/tool/data/config/metric/collectd	0.412s	coverage: 100.0% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/tool/data/config/metric/linux	0.667s	coverage: 91.7% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/tool/data/config/metric/statsd	0.533s	coverage: 100.0% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/tool/data/config/metric/windows	0.659s	coverage: 87.6% of statements
?   	github.com/aws/amazon-cloudwatch-agent/tool/data/interfaze	[no test files]
?   	github.com/aws/amazon-cloudwatch-agent/tool/processors	[no test files]
ok  	github.com/aws/amazon-cloudwatch-agent/tool/processors/agentconfig	0.473s	coverage: 75.0% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/tool/processors/basicInfo	1.274s	coverage: 100.0% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/tool/processors/collectd	0.609s	coverage: 83.3% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/tool/processors/defaultConfig	0.668s	coverage: 78.3% of statements
?   	github.com/aws/amazon-cloudwatch-agent/tool/processors/defaultConfig/advancedPlan	[no test files]
?   	github.com/aws/amazon-cloudwatch-agent/tool/processors/defaultConfig/basicPlan	[no test files]
?   	github.com/aws/amazon-cloudwatch-agent/tool/processors/defaultConfig/standardPlan	[no test files]
ok  	github.com/aws/amazon-cloudwatch-agent/tool/processors/migration	0.730s	coverage: 100.0% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/tool/processors/migration/linux	0.778s	coverage: 85.9% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/tool/processors/migration/windows	0.846s	coverage: 72.7% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/tool/processors/question	0.851s	coverage: 100.0% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/tool/processors/question/events	0.699s	coverage: 86.7% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/tool/processors/question/logs	0.678s	coverage: 87.5% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/tool/processors/question/metrics	0.640s	coverage: 52.5% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/tool/processors/serialization	0.640s	coverage: 100.0% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/tool/processors/ssm	0.720s	coverage: 44.7% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/tool/processors/statsd	0.799s	coverage: 100.0% of statements
?   	github.com/aws/amazon-cloudwatch-agent/tool/processors/template	[no test files]
?   	github.com/aws/amazon-cloudwatch-agent/tool/runtime	[no test files]
?   	github.com/aws/amazon-cloudwatch-agent/tool/stdin	[no test files]
?   	github.com/aws/amazon-cloudwatch-agent/tool/testutil	[no test files]
ok  	github.com/aws/amazon-cloudwatch-agent/tool/util	0.677s	coverage: 52.0% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/translator	0.784s	coverage: 45.4% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/translator/cmdutil	0.699s	coverage: 15.7% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/translator/config	0.734s	coverage: 20.8% of statements
?   	github.com/aws/amazon-cloudwatch-agent/translator/context	[no test files]
ok  	github.com/aws/amazon-cloudwatch-agent/translator/jsonconfig	0.667s	coverage: 59.1% of statements
?   	github.com/aws/amazon-cloudwatch-agent/translator/jsonconfig/mergeJsonRule	[no test files]
?   	github.com/aws/amazon-cloudwatch-agent/translator/jsonconfig/mergeJsonUtil	[no test files]
ok  	github.com/aws/amazon-cloudwatch-agent/translator/toenvconfig	0.690s	coverage: 95.8% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/translator/totomlconfig	1.372s	coverage: 87.5% of statements
?   	github.com/aws/amazon-cloudwatch-agent/translator/translate	[no test files]
ok  	github.com/aws/amazon-cloudwatch-agent/translator/translate/agent	1.350s	coverage: 94.5% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/translator/translate/csm	1.450s	coverage: 83.5% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/translator/translate/globaltags	0.704s	coverage: 75.0% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/translator/translate/logs	2.504s	coverage: 79.2% of statements
?   	github.com/aws/amazon-cloudwatch-agent/translator/translate/logs/logs_collected	[no test files]
?   	github.com/aws/amazon-cloudwatch-agent/translator/translate/logs/logs_collected/files	[no test files]
ok  	github.com/aws/amazon-cloudwatch-agent/translator/translate/logs/logs_collected/files/collect_list	0.687s	coverage: 93.8% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/translator/translate/logs/logs_collected/windows_events	0.745s	coverage: 77.8% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/translator/translate/logs/logs_collected/windows_events/collect_list	0.361s	coverage: 81.0% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/translator/translate/logs/metrics_collected	0.324s	coverage: 46.2% of statements
?   	github.com/aws/amazon-cloudwatch-agent/translator/translate/logs/metrics_collected/ecs	[no test files]
?   	github.com/aws/amazon-cloudwatch-agent/translator/translate/logs/metrics_collected/ecs/cadvisor	[no test files]
?   	github.com/aws/amazon-cloudwatch-agent/translator/translate/logs/metrics_collected/ecs/ec2tagger	[no test files]
?   	github.com/aws/amazon-cloudwatch-agent/translator/translate/logs/metrics_collected/ecs/ecsdecorator	[no test files]
ok  	github.com/aws/amazon-cloudwatch-agent/translator/translate/logs/metrics_collected/emf	0.467s	coverage: 83.9% of statements
?   	github.com/aws/amazon-cloudwatch-agent/translator/translate/logs/metrics_collected/kubernetes	[no test files]
?   	github.com/aws/amazon-cloudwatch-agent/translator/translate/logs/metrics_collected/kubernetes/cadvisor	[no test files]
?   	github.com/aws/amazon-cloudwatch-agent/translator/translate/logs/metrics_collected/kubernetes/ec2tagger	[no test files]
?   	github.com/aws/amazon-cloudwatch-agent/translator/translate/logs/metrics_collected/kubernetes/k8sapiserver	[no test files]
?   	github.com/aws/amazon-cloudwatch-agent/translator/translate/logs/metrics_collected/kubernetes/k8sdecorator	[no test files]
?   	github.com/aws/amazon-cloudwatch-agent/translator/translate/logs/metrics_collected/prometheus	[no test files]
?   	github.com/aws/amazon-cloudwatch-agent/translator/translate/logs/metrics_collected/prometheus/ecsservicediscovery	[no test files]
?   	github.com/aws/amazon-cloudwatch-agent/translator/translate/logs/metrics_collected/prometheus/ecsservicediscovery/dockerlabel	[no test files]
?   	github.com/aws/amazon-cloudwatch-agent/translator/translate/logs/metrics_collected/prometheus/ecsservicediscovery/serviceendpoint	[no test files]
ok  	github.com/aws/amazon-cloudwatch-agent/translator/translate/logs/metrics_collected/prometheus/ecsservicediscovery/taskdefinition	0.541s	coverage: 16.9% of statements
?   	github.com/aws/amazon-cloudwatch-agent/translator/translate/logs/metrics_collected/prometheus/emfprocessor	[no test files]
?   	github.com/aws/amazon-cloudwatch-agent/translator/translate/logs/util	[no test files]
ok  	github.com/aws/amazon-cloudwatch-agent/translator/translate/metrics	0.646s	coverage: 85.4% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/translator/translate/metrics/append_dimensions	0.807s	coverage: 95.3% of statements
?   	github.com/aws/amazon-cloudwatch-agent/translator/translate/metrics/config	[no test files]
ok  	github.com/aws/amazon-cloudwatch-agent/translator/translate/metrics/metric_decoration	0.885s	coverage: 71.0% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/translator/translate/metrics/metrics_collect	0.703s	coverage: 54.3% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/translator/translate/metrics/metrics_collect/agentInternal	0.823s	coverage: 80.0% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/translator/translate/metrics/metrics_collect/collectd	0.760s	coverage: 90.7% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/translator/translate/metrics/metrics_collect/cpu	0.816s	coverage: 90.2% of statements
?   	github.com/aws/amazon-cloudwatch-agent/translator/translate/metrics/metrics_collect/customizedmetrics	[no test files]
ok  	github.com/aws/amazon-cloudwatch-agent/translator/translate/metrics/metrics_collect/disk	0.779s	coverage: 96.6% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/translator/translate/metrics/metrics_collect/diskio	0.803s	coverage: 87.1% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/translator/translate/metrics/metrics_collect/ethtool	0.905s	coverage: 84.4% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/translator/translate/metrics/metrics_collect/mem	0.965s	coverage: 89.3% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/translator/translate/metrics/metrics_collect/net	0.932s	coverage: 82.9% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/translator/translate/metrics/metrics_collect/netstat	0.919s	coverage: 80.0% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/translator/translate/metrics/metrics_collect/processes	0.939s	coverage: 81.8% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/translator/translate/metrics/metrics_collect/procstat	0.924s	coverage: 96.7% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/translator/translate/metrics/metrics_collect/statsd	0.961s	coverage: 90.9% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/translator/translate/metrics/metrics_collect/swap	0.957s	coverage: 85.0% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/translator/translate/metrics/rollup_dimensions	0.969s	coverage: 89.7% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/translator/translate/metrics/util	0.902s	coverage: 26.9% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/translator/translate/util	0.851s	coverage: 26.4% of statements
ok  	github.com/aws/amazon-cloudwatch-agent/translator/util	0.819s	coverage: 5.4% of statements
?   	github.com/aws/amazon-cloudwatch-agent/translator/util/ec2util	[no test files]
?   	github.com/aws/amazon-cloudwatch-agent/translator/util/ecsutil	[no test files]
?   	github.com/aws/amazon-cloudwatch-agent/translator/util/httpclient	[no test files]
```

**Pull request merged**
